apps/shell/app.js

```diff
@@ -46,16 +46,17 @@ var wss = new WebSocket.Server({ server: server });
 
 wss.on('connection', function connection (ws, req) {
   var match;
-  var host = process.env.DEFAULT_SSHHOST || 'localhost';
+  var host = process.env.DEFAULT_SSHHOST;
   var dir;
   var term;
   var cmd, args;
 
   console.log('Connection established');
 
+  var whitelist = process.env.SSHHOST_WHITELIST.split(':')
   // Determine host and dir from request URL
   if (match = req.url.match(process.env.PASSENGER_BASE_URI + '/ssh/([^\\/]+)(.+)?$')) {
-    if (match[1] !== 'default') host = match[1];
+    if (match[1] !== 'default'&& whitelist.includes(match[1])) host = match[1];
```

Comment:
This is the wrong place to do this test. This line was just determining what host the client is requesting to connect to.

After it has been determined what host to connect to, that host should be checked against the whitelist, and if not allowed, the connection should be closed, prior to calling `pty.spawn`.